src/crate/testing/layer.py

```diff
@@ -269,6 +272,10 @@ def start(self):
             # this is necessary if no static port is assigned
             self.http_url = wait_for_http_url(self.process.stdout, verbose=self.verbose)
 
+        self._out_thread = threading.Thread(target=self._consume, args=(self.process.stdout,))
```

Comment:
Could also add a comment here that this is to unblock the process.stdout buffer from filling up, which would cause the process to block